scene/gui/line_edit.cpp

```diff
@@ -156,449 +283,155 @@ void LineEdit::_gui_input(Ref<InputEvent> p_event) {
-						update_cached_width();
-						set_cursor_position(0);
-						_text_changed();
-					}
-
-				} break;
-
-				case (KEY_Y): { // PASTE (Yank for unix users).
-
-					if (editable) {
-						paste_text();
-					}
-
-				} break;
-				case (KEY_K): { // Delete from cursor_pos to end.
-
-					if (editable) {
-						deselect();
-						text = text.substr(0, cursor_pos);
-						_text_changed();
-					}
-
-				} break;
-				case (KEY_A): { // Select all.
-					select();
-
-				} break;
-#ifdef APPLE_STYLE_KEYS
-				case (KEY_LEFT): { // Go to start of text - like HOME key.
-					shift_selection_check_pre(k->get_shift());
-					set_cursor_position(0);
-					shift_selection_check_post(k->get_shift());
-				} break;
-				case (KEY_RIGHT): { // Go to end of text - like END key.
-					shift_selection_check_pre(k->get_shift());
-					set_cursor_position(text.length());
-					shift_selection_check_post(k->get_shift());
-				} break;
-				case (KEY_BACKSPACE): {
-					if (!editable)
-						break;
-
-					// If selected, delete the selection
-					if (selection.enabled) {
-						selection_delete();
-						break;
-					}
-
-					// Otherwise delete from cursor to beginning of text edit
-					int current_pos = get_cursor_position();
-					if (current_pos != 0) {
-						delete_text(0, current_pos);
-					}
-				} break;
-#endif
-				default: {
-					handled = false;
-				}
-			}
-
-			if (handled) {
-				accept_event();
-				return;
+		// Default is ENTER, KP_ENTER. Cannot use ui_accept as default includes SPACE
+		if (k->is_action("ui_text_newline")) {
+			emit_signal("text_entered", text);
+			if (DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_VIRTUAL_KEYBOARD) && virtual_keyboard_enabled) {
+				DisplayServer::get_singleton()->virtual_keyboard_hide();
 			}
 		}
 
-		_reset_caret_blink_timer();
-		if (!k->get_metakey()) {
-			bool handled = true;
-			switch (code) {
-				case KEY_KP_ENTER:
-				case KEY_ENTER: {
-					emit_signal("text_entered", text);
-					if (DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_VIRTUAL_KEYBOARD) && virtual_keyboard_enabled) {
-						DisplayServer::get_singleton()->virtual_keyboard_hide();
-					}
-
-				} break;
-
-				case KEY_BACKSPACE: {
-					if (!editable) {
-						break;
-					}
-
-					if (selection.enabled) {
-						selection_delete();
-						break;
-					}
-
-#ifdef APPLE_STYLE_KEYS
-					if (k->get_alt()) {
-#else
-					if (k->get_alt()) {
-						handled = false;
-						break;
-					} else if (k->get_command()) {
-#endif
-						int cc = cursor_pos;
-						bool prev_char = false;
-
-						while (cc > 0) {
-							bool ischar = _is_text_char(text[cc - 1]);
-
-							if (prev_char && !ischar) {
-								break;
-							}
-
-							prev_char = ischar;
-							cc--;
-						}
-
-						delete_text(cc, cursor_pos);
-
-						set_cursor_position(cc);
+		if (is_shortcut_keys_enabled()) {
+			if (k->is_action("ui_copy")) {
+				copy_text();
+				event_accepted = true;
```

Comment:
Would it make sense to call `accept_event` and return, rather then having `event_accepted` bool as it no longer a switch. Ideally it shouldn't need to evaluate every shortcut if we find a match. Same goes for `TextEdit`.

Reply:
Yeah, I think this is a remnant of some early work on the PR. I think accepting the event and returning should be fine.